dockerd-entrypoint.sh

```diff
@@ -41,7 +41,10 @@ _tls_generate_certs() {
 		_tls_ensure_private "$dir/ca/key.pem"
 		openssl req -new -key "$dir/ca/key.pem" \
 			-out "$dir/ca/cert.pem" \
-			-subj '/CN=docker:dind CA' -x509 -days "$certValidDays"
+			-subj '/CN=docker:dind CA' \
+			-x509 \
+			-days "$certValidDays" \
+			-addext keyUsage=critical,digitalSignature,keyCertSign
```

Comment:
All the OpenSSL docs say about `critical` here is this:

> If critical is present then the extension will be marked as critical.

Do you have a reference for why this is necessary? :thinking:

The docs also suggest this can be more whitespace-y, so if we need to touch this again I think I might try something like this:

```suggestion
			-addext 'keyUsage = critical, digitalSignature, keyCertSign'
```

But we should probably add a new test or update our existing test to make sure this stays working before we touch it again (since you've done good validation on this change as-is). :heart:

Reply:
From the [RFC section 4.2.1.3 Key Usage](https://datatracker.ietf.org/doc/html/rfc3280#section-4.2.1.3):

> This extension MUST appear in certificates that contain public keys
   that are used to validate digital signatures on other public key
   certificates or CRLs.  When this extension appears, it SHOULD be
   marked critical.